app/src/test/java/waiter/RouterTest.java

```diff
@@ -29,4 +29,40 @@ void simpleGetWithBodyRoute() {
         assertEquals("Content-Length: 11", route.getHeaders());
         assertEquals("Hello world", route.getBody());
     }
+
+    @Test
+    void headSimpleGetReturnsOKRouteWithoutBody() {
+        Request request = new Request("HEAD","/simple_get", "HTTP/1.1");
+
+        Route route = new Router().getRoute(request);
+
+        assertEquals("HTTP/1.1", route.getProtocol());
+        assertEquals("200 OK", route.getStatus());
+        assertEquals("Content-Length: 0", route.getHeaders());
+        assertEquals("", route.getBody());
+    }
+
+    @Test
+    void NoMethodOrUrlFoundReturnsInternalServerErrorRouteWithoutBody() {
```

Comment:
I'm not sure this is quite right. If the route is not found, then it should be 404. If the route exists, but the method is not supported then we should return a [405](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) instead.

Reply:
That is true, but I thought this was testing if both the route and the method are not supported

Reply:
In that case, if a route is not found, regardless of method, we should always return 404 instead of a 500

Reply:
For example, you can try in Postman, doing a `POST` to `google.com/bar` and it'll return a 404.

Reply:
Addressed in [this commit](https://github.com/jonmanahan/waiter/pull/17/commits/bf3be9d6f0dac707cde56cd7b18a4485db55b49b)